.travis.yml

```diff
@@ -31,14 +31,15 @@ before_install:
   - if [[ $JDK_FOR_TEST == "oraclejdk9" ]]; then sudo dpkg -i deps/oracle-java9-installer.deb; fi
 
 install:
-  - if [[ ! -e deps/eclipse.tar.gz ]]; then wget 'https://www.eclipse.org/downloads/download.php?file=/eclipse/downloads/drops4/R-4.6.3-201703010400/eclipse-SDK-4.6.3-linux-gtk-x86_64.tar.gz&r=1' -O deps/eclipse.tar.gz; fi
-  - tar xzvf deps/eclipse.tar.gz eclipse
+  - if [[ ! -e deps/eclipse-SDK-4.6.3-linux-gtk-x86_64.tar.gz ]]; then wget 'https://www.eclipse.org/downloads/download.php?file=/eclipse/downloads/drops4/R-4.6.3-201703010400/eclipse-SDK-4.6.3-linux-gtk-x86_64.tar.gz&r=1' -O deps/eclipse-SDK-4.6.3-linux-gtk-x86_64.tar.gz; fi
+  - rm -rf eclipse
+  - tar xzvf deps/eclipse-SDK-4.6.3-linux-gtk-x86_64.tar.gz eclipse
   - echo eclipseRoot.dir=$(pwd)/eclipse > eclipsePlugin/local.properties
 
 script:
   # Manually switch to JDK9 if needed
   - if [[ $JDK_FOR_TEST == "oraclejdk9" ]]; then remove_dir_from_path $JAVA_HOME/bin; export JAVA_HOME=/usr/lib/jvm/java-9-oracle; export JAVA_HOME=/usr/lib/jvm/java-9-oracle; fi
-  - ./gradlew build smoketest sonarqube -S
+  - ./gradlew clean build smoketest sonarqube -S
```

Comment:
Then I think this `clean` cannot ensure what you need, because `build` dir which is cleared by `clean` is already not cached (non incremental).

If you know which Eclipse data was cached, it is better to configure [before_cache phase](https://docs.travis-ci.com/user/caching/#before_cache-phase) to delete things which should not be cached.